api/webhook.go

```diff
@@ -792,11 +737,98 @@ func publishToQueue(queue queue.Service, db database.Service, p *pipeline.Build,
 	}
 }
 
+func handleRepositoryEvent(c *gin.Context, m *types.Metadata, h *library.Hook, r *library.Repo) (*library.Repo, error) {
+	logrus.Debugf("webhook is repository event, making necessary updates to repo %s", r.GetFullName())
+
+	defer func() {
+		// send API call to update the webhook
+		err := database.FromContext(c).CreateHook(h)
+		if err != nil {
+			logrus.Errorf("unable to create webhook %s/%d: %v", r.GetFullName(), h.GetNumber(), err)
+		}
+	}()
+
+	switch h.GetEventAction() {
+	// if action is rename, go through rename routine
+	case constants.ActionRenamed:
+		r, err := renameRepository(h, r, c, m)
+		if err != nil {
+			h.SetStatus(constants.StatusFailure)
+			h.SetError(err.Error())
+
+			return nil, err
+		}
+
+		return r, nil
+	// if action is archived, unarchived, or edited, perform edits to relevant repo fields
+	case "archived", "unarchived", constants.ActionEdited:
+		// send call to get repository from database
```

Comment:
suggestion: add log entry. the first thing `renameRepository` does for rename event, is log a debug message, we should do the same here.

Reply:
Sure